### PR TITLE
New Subworkflow: `fastq_ena_prep_adapterremoval_md5sum`

### LINKED ISSUE #3294: new subworkflow: FASTQ/ENA_PREP/ADAPTERREMOVAL
### Is there an existing subworkflow for this?

- [X] I have searched for the existing subworkflow

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this subworkflow, I added myself to the `Assignees` to facilitate tracking who is working on the subworkflow

### PR DESCRIPTION
**Pull request by TCLamnidis**
This reimplements [maxibor/enaprep](https://github.com/maxibor/enaprep) into a DSL2 subworkflow.

First, adapters are trimmed from FastQ files without any filtering or collapsing using AdapterRemoval. 
Then, the md5sums of the different trimmed FastQ files are calculated, collated and published in a file named `checksums.md5`.

## PR checklist

Closes #3294 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by TCLamnidis** [comment]
- Should I rename the SWF to `_adapterremoval_md5sum`?

File: subworkflows/nf-core/fastq_ena_prep_adapterremoval/nextflow.config [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// IMPORTANT: Add this configuration to your modules.config
+
+process {
+    withName: '.*:FASTQ_ENA_PREP_ADAPTERREMOVAL:ENA_PREP_ADAPTERREMOVAL' {
+        ext.args = { "--qualitybase ${meta.phred}" }
+    }
```
Comment (jfy133) [inline subworkflows/nf-core/fastq_ena_prep_adapterremoval/nextflow.config:6]: Make sure all other quality/and or lenght filtering is turned off: https://ena-docs.readthedocs.io/en/latest/submit/fileprep/reads.html#fastq-format

**Review by jfy133 — CHANGES_REQUESTED** [review]
I agree to the renaming to make it more 'general' but otherwise looks OK

File: subworkflows/nf-core/fastq_ena_prep_adapterremoval/main.nf [hunk 2]
```diff
@@ -0,0 +1,73 @@
+include { ADAPTERREMOVAL as ENA_PREP_ADAPTERREMOVAL } from '../../../modules/nf-core/adapterremoval/main'
+include { MD5SUM                                    } from '../../../modules/nf-core/md5sum/main'
+include { CAT_CAT                                   } from '../../../modules/nf-core/cat/cat/main'
+
+workflow FASTQ_ENA_PREP_ADAPTERREMOVAL {
+
+    take:
+    ch_reads        // channel: [ val(meta), [ fastq ], val(phred_plus) ]
```
[the remark this answers is not in the document]
Comment (TCLamnidis) [reply subworkflows/nf-core/fastq_ena_prep_adapterremoval/main.nf:8]: in 90% of cases I would not expect a mix, but why limit the SWF, when a dev will need to do very minor channel tweaks to add that field within the channel?

File: subworkflows/nf-core/fastq_ena_prep_adapterremoval/meta.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+name: "fastq_ena_prep_adapterremoval"
+description: Trim adapters from FastQs to ENA specification, and calculate checksums.
```
[the remark this answers is not in the document]
Comment (TCLamnidis) [reply subworkflows/nf-core/fastq_ena_prep_adapterremoval/meta.yml]: Added that link to SWF description

**Comment by TCLamnidis** [comment]
Renamed:
`fastq_ena_prep_adapterremoval` -> `fastq_ena_prep_adapterremoval_md5sum`

**Pull request closed without merging**